src/test/ui/hygiene/prelude-import-hygiene.rs

```diff
@@ -0,0 +1,29 @@
+// Make sure that attribute used when injecting the prelude are resolved
+// hygienically.
+
+// check-pass
+// aux-build:not-libstd.rs
+
+//revisions: rust2015 rust2018
+//[rust2018] edition:2018
+
+// The prelude import shouldn't see these as candidates for when it's trying to
+// use the built-in macros.
+extern crate core;
+use core::prelude::v1::test as prelude_import;
+use core::prelude::v1::test as macro_use;
+
+// Should not be used for the prelude import - not a concern in the 2015 edition
+// because `std` is already declared in the crate root.
+#[cfg(rust2018)]
+extern crate not_libstd as std;
```

Comment:
Well, this is kinda arguable.
If we can reroute standard library (including prelude imports) to somewhere else with `--extern std=/path/to/my_std`, then why can't we do the same thing in source code with `extern crate my_std as std`.
cc https://github.com/rust-lang/rust/issues/63687#issuecomment-526241118

Reply:
I see it the other way `--extern std=/path/to/my_std` means that there doesn't need to be a way to override `std` in source code, especially one that only works on the 2018 edition.

Looking at the linked issue, the crate in question isn't `#![no_std]` and uses the 2015 edition, so it's not overriding anything there.